viam_cartographer.go

```diff
@@ -270,6 +271,20 @@ func New(
 		}
 	}()
 
+	// if we have an existing map, check if there is an edited map within the package
+	if cartoSvc.existingMap != "" {
+		packageDir := filepath.Dir(svcConfig.ExistingMap)
+
+		packageDir = filepath.Clean(packageDir + editedMapName)
+		bytes, err := os.ReadFile(packageDir)
+		if err != nil && !errors.Is(err, os.ErrNotExist) {
+			return nil, err
+		}
+		if len(bytes) > 0 {
+			cartoSvc.editedMap = &bytes
+		}
+	}
```

Comment:
so theres not really a good way to do this now, but at some point we should probably add a function to do this in rdk/slam supported sdks. Other option is documenting how slam-map packages workso they know how to do this.
cc @randhid

Reply:
tbh i wonder if there was a way to have the sdk/server codes pull from the packages directly rather than interacting with the slam interface, but thats def a question for another day

Reply:
We can expose helpers client side on the different sdks. However you're right it's heavily dependent on the implementer to implement. Which is fine I think.

Reply:
I'd keep it carto-side fro now becuase we have no idea how this API works or behaves with visual slam and whether we need a non-pcd expansion for those maps.

.... or bump sensor slam.